### PR TITLE
Updates for pycodestyle.

### LINKED ISSUE #2974: Pycodestyle update causing our lint to fail.
From what I can tell, there are three new errors that pycodestyle is finding.

E306: expected 1 blank line before a nested definition, found 0
~E123: closing bracket does not match indentation of opening bracket's line~
~E722: do not use bare except~

See: #2973

### PR DESCRIPTION
**Pull request by daspecster**
These changes allow pycodestyle to pass.

Closes #2974

### CODE REVIEW

File: bigquery/unit_tests/test_dataset.py [hunk 1]
```diff
@@ -838,7 +838,7 @@ def api_request(self, **kw):
 
         try:
             response, self._responses = self._responses[0], self._responses[1:]
-        except:
+        except Exception:
```
Comment (dhermes) [inline bigquery/unit_tests/test_dataset.py]: :-1: Never catch `Exception`. Here the exception is an `IndexError`:

```python
>>> a = []
>>> a[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
```

File: core/unit_tests/test__helpers.py [hunk 2]
```diff
@@ -138,12 +138,14 @@ def _call_fut(self, project=None):
         return _determine_default_project(project=project)
 
     def test_it(self):
+
```
Comment (dhermes) [inline core/unit_tests/test__helpers.py]: `pycodestyle` complained about this?

**Comment by tseaver** [comment]
Are the whitespace changes actually required by the tool, or did you just add them because they seemed needed?  I know of no justification in PEP8 etc. for them.

File: core/unit_tests/test__helpers.py [hunk 3]
```diff
@@ -138,12 +138,14 @@ def _call_fut(self, project=None):
...
         with mock.patch('google.auth.default', autospec=True) as default:
             default.return_value = (
                 mock.sentinel.credentials, mock.sentinel.project)
             project = self._call_fut()
 
         self.assertEqual(project, mock.sentinel.project)
+
```
Comment (dhermes) [inline core/unit_tests/test__helpers.py]: Also surprised `pycodestyle` said anything here

**Comment by dhermes** [comment]
> Are the whitespace changes actually required by the tool, or did you just add them because they seemed needed? I know of no justification in PEP8 etc. for them.

Jinx, bigly

**Comment by daspecster** [comment]
@tseaver I'm wondering if it's a bug in pycodestyle. I moved them around a little bit to try and see what it was looking for and that was the way it had to be to pass. We could probably disable that error.

**Comment by dhermes** [comment]
@daspecster Please provide the errors so we can try to help

**Comment by daspecster** [comment]
There were only two types of errors. One was the bare except and this was the other one.

```
core/unit_tests/test__helpers.py:147:9: E306 expected 1 blank line before a nested definition, found 0
```

**Comment by tseaver** [comment]
I vote to suppress that error with prejudice.

**Comment by dhermes** [comment]
I'd agree, we should suppress, but not before filing an issue with the `pycodestyle` folks

**Comment by daspecster** [comment]
SGTM, I'll update and I can file the issue on pycodestyle as well.

**Comment by daspecster** [comment]
There's a bare except in [transaction.py](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/3584917bc96ce2574d700d40b66eeacb3a949043/datastore/google/cloud/datastore/transaction.py#L189) that I'm not sure what the possible exception types could be. I think it could be a few things. If there is a known exception type for this then I can stop ignoring E306.

**Comment by dhermes** [comment]
The except in `transaction.py` is meant to be bare (it's somewhat of a special case). You can locally disable `pycodestyle` checks via `noqa` ([e.g.][1])

[1]: https://github.com/dhermes/bezier/blob/6e44c92f0afdcb14e729dccdf40a34aaada7b9d4/src/bezier/_curve_helpers.py#L51

File: datastore/google/cloud/datastore/transaction.py [hunk 4]
```diff
@@ -186,7 +186,7 @@ def begin(self):
         try:
             self._id = self._client._connection.begin_transaction(
                 self.project)
-        except:
+        except:  # noqa: E722
```
Comment (dhermes) [inline datastore/google/cloud/datastore/transaction.py]: @daspecster Still applies

Comment (daspecster) [inline datastore/google/cloud/datastore/transaction.py]: https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2973/commits/aeeead940047fd64c31f34b9b8c2ce0be36f83bf

File: datastore/google/cloud/datastore/transaction.py [hunk 5]
```diff
@@ -186,7 +186,7 @@ def begin(self):
         try:
             self._id = self._client._connection.begin_transaction(
                 self.project)
-        except:
+        except:  # noqa: E722 do not use bare except, specify exception instead
```
Comment (dhermes) [inline datastore/google/cloud/datastore/transaction.py]: FWIW I assumed you'd summarize in your own words, but this fits on one line, so it's fine

Comment (daspecster) [inline datastore/google/cloud/datastore/transaction.py]: Yeah, but that text is searchable.  e.g. if you copy and paste that into google you'll get more information and the source of the definition.

**Comment by lukesneeringer** [comment]
> The except in transaction.py is meant to be bare (it's somewhat of a special case). You can locally disable pycodestyle checks via noqa (e.g.)

Or `except Exception`, if it passes the linter.

**Comment by dhermes** [comment]
@lukesneeringer I am allergic to `except Exception` **and** that misses nasty exception types that only inherit from `BaseException` **and** `pylint` will complain about `Exception` being too broad

**Comment by lukesneeringer** [comment]
Fair on all counts. :-)

**Pull request merged**